From 7eeefe3484081b72a0ad20a2b788e3f514da0e18 Mon Sep 17 00:00:00 2001
From: Ashley Hindle <hey@ashleyhindle.com>
Date: Thu, 23 Mar 2023 11:37:18 +0000
Subject: [PATCH] Hopefully fix duplicate carddav issue

---
 app/Services/VCard/ImportVCard.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/Services/VCard/ImportVCard.php b/app/Services/VCard/ImportVCard.php
index 26dde817e9d..071bca1c6be 100644
--- a/app/Services/VCard/ImportVCard.php
+++ b/app/Services/VCard/ImportVCard.php
@@ -575,7 +575,7 @@ private function importGeneralInformation(?Contact $contact, VCard $entry): Cont
         $contactData = $this->importGender($contactData, $entry);
         $contactData = $this->importBirthday($contactData, $entry);
 
-        if ($contact !== null && $contactData !== $original) {
+        if ($contact !== null || $contactData !== $original) {
             $contact = app(UpdateContact::class)->execute($contactData);
         } else {
             $contact = app(CreateContact::class)->execute($contactData);

